From fc83250b8c6186d0d620b1bbeef073255c650e93 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:20:22 +0200
Subject: [PATCH] rule: constify set_is_non_concat_range()

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 0ff245fef0cff92651e464dd2d282f8d18fbfbc3

commit 0ff245fef0cff92651e464dd2d282f8d18fbfbc3
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Fri Jan 3 17:40:56 2025 +0100

    rule: constify set_is_non_concat_range()

    This is read-only, constify it.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 include/rule.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/rule.h b/include/rule.h
index 7b6551a..1a851ff 100644
--- a/include/rule.h
+++ b/include/rule.h
@@ -420,7 +420,7 @@ static inline bool set_is_interval(uint32_t set_flags)
 	return set_flags & NFT_SET_INTERVAL;
 }
 
-static inline bool set_is_non_concat_range(struct set *s)
+static inline bool set_is_non_concat_range(const struct set *s)
 {
 	return (s->flags & NFT_SET_INTERVAL) && s->desc.field_count <= 1;
 }
